src/Components/Single Article/Comments/AddComment.jsx

```diff
@@ -0,0 +1,54 @@
+import { useContext, useEffect, useState } from "react";
+import { UserContext } from "../../Context/UserContext";
+import { postComment } from "../../../Utils/postComment";
+
+const AddComment = ({ id, setComments }) => {
+  const [commentData, setCommentData] = useState("");
+
+  const userContext = useContext(UserContext);
+
+  const onChangeHandler = (event) => {
+    setCommentData(event.target.value);
+  };
+
+  const onSubmitHandler = (event) => {
+    event.preventDefault();
+    setComments((curr) => [
+      {
+        article_id: id,
+        author: userContext.user,
+        body: commentData,
+        comment_id: curr.length + 1,
+        created_at: new Date().toString(),
+        votes: 0,
+      },
+
+      ...curr,
+    ]);
+
+    postComment({ username: userContext.user, comment: commentData }, id).catch(
```

Comment:
we also want to remove the optimistically added comment here in the catch block if we fail to add it to the db